### DISCUSSION TITLE
What are valid values for "image"?

### DISCUSSION THREAD

**Discussion by rptb1** [cirruslabs/cirrus-ci-docs#1147] [Q&A] [answered]
I'm investigating Cirrus CI as a replacement for Travis CI for build and test of the [MPS](https://github.com/Ravenbrook/mps).  You can see my experiment in [this pull request](https://github.com/Ravenbrook/mps/pull/145).

I'm finding the Cirrus documentation a bit light on detail.  Mainly I do not know how to determine the valid values for `image` and what they mean.

For [Linux](https://cirrus-ci.org/guide/linux/) `arm_container` I can get an image of `gcc:latest` but not `clang:latest`.  How could I check what's available?  What does this `gcc:latest` string refer to?

For a [macOS VM](https://cirrus-ci.org/guide/macOS/), the docs show `image` as `ghcr.io/cirruslabs/macos-ventura-base:latest`.  Why is this a path?  Why this particular path?

For [FreeBSD VMs](https://cirrus-ci.org/guide/FreeBSD/), `image` becomes `image_family`.  Why?  What makes `freebsd_13_0` a "family" of images?

Our work is quite platform sensitive (includes assembly language, for example) and so the details are important to us.

Many thanks.

**Comment by edigaryev** [comment, marked as answer]
> For [Linux](https://cirrus-ci.org/guide/linux/) `arm_container` I can get an image of `gcc:latest` but not `clang:latest`. How could I check what's available? What does this `gcc:latest` string refer to?

This string refers to the container image that you'd normally use when running containers with [Docker](https://www.docker.com/) or any other container engine, for example:

```
docker run -it --rm gcc:latest
```

You can specify a full path like `docker.io/gcc:latest` (or use other container registry like `ghcr.io/...`) or use a shorter syntax, which defaults to the images available on https://hub.docker.com/.

Docker Hub currently [has official `gcc` image](https://hub.docker.com/_/gcc), but [no official `clang` image](https://hub.docker.com/_/clang).

>For a [macOS VM](https://cirrus-ci.org/guide/macOS/), the docs show image as ghcr.io/cirruslabs/macos-ventura-base:latest. Why is this a path? Why this particular path?

It's also a container image name (though, only compatible with [Tart virtualization toolset](https://github.com/cirruslabs/tart)), with the exception that for now [only Cirrus CI-managed images](https://github.com/orgs/cirruslabs/packages) are allowed.

As for why this particular path — it's just the latest macOS image available at the moment and we update it periodically when the new major macOS version gets released.

> For [FreeBSD VMs](https://cirrus-ci.org/guide/FreeBSD/), `image` becomes `image_family`. Why? What makes `freebsd_13_0` a "family" of images?

See "Under the Hood" note [in the FreeBSD VMs documentation](https://cirrus-ci.org/guide/FreeBSD/#freebsd-virtual-machines), it explains that FreeBSD VMs are just a mapping for running GCE instances, which require an image family to locate the image.

You can find a list of available images and their families [using the following command](https://cloud.google.com/compute/docs/images#freebsd):

```
% gcloud compute images list --project freebsd-org-cloud-dev --no-standard-images | head -n 10
NAME                                     PROJECT                FAMILY             DEPRECATED  STATUS
freebsd-11-3-stable-amd64-v20190801      freebsd-org-cloud-dev  freebsd-11-3-snap              READY
freebsd-11-3-stable-amd64-v20191212      freebsd-org-cloud-dev  freebsd-11-3-snap              READY
freebsd-11-3-stable-amd64-v20191219      freebsd-org-cloud-dev  freebsd-11-3-snap              READY
freebsd-11-3-stable-amd64-v20191226      freebsd-org-cloud-dev  freebsd-11-3-snap              READY
freebsd-11-3-stable-amd64-v20200102      freebsd-org-cloud-dev  freebsd-11-3-snap              READY
freebsd-11-3-stable-amd64-v20200109      freebsd-org-cloud-dev  freebsd-11-3-snap              READY
freebsd-11-3-stable-amd64-v20200116      freebsd-org-cloud-dev  freebsd-11-3-snap              READY
freebsd-11-3-stable-amd64-v20200123      freebsd-org-cloud-dev  freebsd-11-3-snap              READY
freebsd-11-3-stable-amd64-v20200130      freebsd-org-cloud-dev  freebsd-11-3-snap              READY
```
